### PR TITLE
Allow for bare clone and retrieving commit history

### PR DESCRIPTION
**Pull request by awanlin**
Signed-off-by: Andre Wanlin <awanlin@rapidrtc.com>

## Hey, I just made a Pull Request!

These changes allow for a bare git clone which gives you just the git history in the `.git` folder and also adds a git log function to get commit history

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by awanlin** [comment]
Sorry there are no tests for this, happy to add if you have some advice on how I would do that

File: .changeset/few-plums-enjoy.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/backend-common': patch
+---
+
+Updated Git to allow for bare clone
```
Comment (jhaals) [inline .changeset/few-plums-enjoy.md:5]:
```suggestion
Adds `depth` and `noCheckout` options to Git clone
```

Comment (jhaals) [reply .changeset/few-plums-enjoy.md:5]: Some text about adding `log` would be good to :)

File: packages/backend-common/src/scm/git.ts [hunk 2]
```diff
@@ -76,12 +76,15 @@ export class Git {
     return git.commit({ fs, dir, message, author, committer });
   }
 
+  // https://isomorphic-git.org/docs/en/clone
```
Comment (jhaals) [inline packages/backend-common/src/scm/git.ts:79]: Replace `//` comments with a docstrings `/**`, that will also remove the undocumented warning in api-report :)

**Review by jhaals — COMMENTED** [review]
Nice work! 👍 

There's a test failing in https://github.com/backstage/backstage/blob/master/packages/backend-common/src/scm/git.test.ts
That's likely the place where you want to add some new tests for `log` as well.

**Comment by awanlin** [comment]
@jhaals thanks for the review, I'll get working on this. It's odd though, never got an email about this 🤔

File: .changeset/few-plums-enjoy.md (hunk 1 above)
Comment (awanlin) [reply .changeset/few-plums-enjoy.md]: Totally forgot to add details on that, I've updated, should be good now :)

File: packages/backend-common/src/scm/git.ts (hunk 2 above)
Comment (awanlin) [reply packages/backend-common/src/scm/git.ts]: 👍 I've switched these all over

**Comment by awanlin** [comment]
Ok, got the test fixed, add on for the new `log` function. Can you give this another look over, please? @jhaals 

It would be awesome if I could get this into the Jan 4th release 🤞

File: packages/backend-common/src/scm/git.ts [hunk 3]
```diff
@@ -90,7 +93,8 @@ export class Git {
       dir,
       ref,
       singleBranch: true,
-      depth: 1,
+      depth: depth ?? 1,
+      noCheckout: noCheckout ?? false,
```
Comment (jhaals) [inline packages/backend-common/src/scm/git.ts:97]:
```suggestion
      noCheckout,
```

This should work as well and would remove the need to assert for noCheckout false in the test above.

**Comment by jhaals** [comment]
@awanlin I took the liberty to address the changes myself in case you are in a different time zone so that this still make it into the release :)

**Review by freben — APPROVED** [review]

**Review by benjdlambert — APPROVED** [review]

**Comment by awanlin** [comment]
@jhaals thanks! I often second guess myself around truthy/falsey in JS/TS so I ten to just put it in explicitly. Totally cool with your suggestion and thanks for making it. Working on a plugin that will use this but want to give it a few cycles internally before contributing it

**Pull request merged**